File: 1837 B. Comparison String/Solution.cpp
// 319949236	        May/17/2025 03:47UTC+6	        Krishno_MD	        1837B - Comparison String	        C++20 (GCC 13-64)	        Accepted	        46 ms	        0 KB

#include<bits/stdc++.h>
using namespace std;
//Krishno_Mondol_KUET_CSE 
#define ll long long 
void solve(){
    int n; cin>>n;
    string str; cin>>str;
    int maxx = 1, s_count = 1;
    for(int i=1; i<str.size(); i++){
        if(str[i]==str[i-1]){s_count++; maxx = max(maxx,s_count);}
        else{maxx = max(maxx,s_count); s_count = 1;}
    }
    cout<<maxx+1<<endl;
}
int main(){ int t; cin>>t; while(t--){ solve(); } return 0;}

File: 1837 B. Comparison String/Solution.py
# 319949448	        May/17/2025 03:52UTC+6	        Krishno_MD	        1837B - Comparison String	        PyPy 3-64	        Accepted	        109 ms	        1800 KB

# Krishno_Mondol_KUET_CSE

def solution():
    n = int(input())
    str = input()
    maxx,b_count = 1,1
    for i in range(1,len(str)):
        if str[i]==str[i-1]:
            b_count += 1
            maxx = max(maxx,b_count)
        else:
            maxx = max(maxx,b_count)
            b_count = 1
    print(maxx+1)

test = int(input())
for i in range(test):
    solution()
